cli/azd/pkg/tools/azcli/azcli.go

```diff
@@ -62,8 +63,8 @@ type AzCli interface {
 	ListExtensions(ctx context.Context) ([]AzCliExtensionInfo, error)
 	GetCliConfigValue(ctx context.Context, name string) (AzCliConfigValue, error)
 	GetSubscriptionTenant(ctx context.Context, subscriptionId string) (string, error)
-	GetSubscriptionDeployment(ctx context.Context, subscriptionId string, deploymentName string) (AzCliDeployment, error)
-	GetResourceGroupDeployment(ctx context.Context, subscriptionId string, resourceGroupName string, deploymentName string) (AzCliDeployment, error)
+	GetSubscriptionDeployment(ctx context.Context, subscriptionId string, deploymentName string) (*armresources.DeploymentExtended, error)
+	GetResourceGroupDeployment(ctx context.Context, subscriptionId string, resourceGroupName string, deploymentName string) (*armresources.DeploymentExtended, error)
```

Comment:
@wbreza What's the strategy around structures exposed? Are we reusing SDK structs or creating our own structs?

Reply:
The current strategy was to use the current structures defined in the `azcli` package for the initial implementation swap.  Then we can re-visit if these are still required when we split out the `azcli` package into more modular components.

For @vhvb1989's use case the structure was sufficiently complex to just warrant using the SDK model.

Reply:
yeah, the structure is significantly large to create the translated copy into the actual model. 
I considered getting the http raw response from the sdk client to parse into the actual model, but the sdk client would create the sdk model anyway